apiserver/registration.go

```diff
@@ -172,17 +169,18 @@ func (h *registerUserHandler) processPost(
 	return userTag, response, nil
 }
 
-func getConnectorInfoer(ctx context.Context, model stateenvirons.Model, cloudService common.CloudService, credentialService common.CredentialService) (environs.ConnectorInfo, error) {
-	configGetter := stateenvirons.EnvironConfigGetter{
-		Model: model, CloudService: cloudService, CredentialService: credentialService}
-	environ, err := common.EnvironFuncForModel(model, cloudService, credentialService, configGetter)(ctx)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if connInfo, ok := environ.(environs.ConnectorInfo); ok {
-		return connInfo, nil
+func getConnectorInfoer(
+	ctx context.Context,
+	providerFactory providertracker.ProviderFactory,
+	controllerModelID model.UUID,
+) (environs.ConnectorInfo, error) {
+
+	type connectableProvider interface {
+		environs.BootstrapEnviron
```

Comment:
I may be missing something, but is the `BootstrapEnviron` actually needed here? Its not needed by the return type of this function